util/collate/collate.go

```diff
@@ -39,6 +40,15 @@ func GetCollator(collate string) Collator {
 	return ctor
 }
 
+// GetCollatorByID get the collator according to id, it will return the binary collator if the corresponding collator doesn't exist.
+func GetCollatorByID(id int) Collator {
```

Comment:
I found that, there's some collation information in parser's `mysql/charset.go`. Should we uniform them?